### PR TITLE
feat(fluentd-modsecurity): add fluentd-modsecurity docker image

### PR DESCRIPTION
**Pull request by ylascombe**

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -19,6 +19,7 @@ include:
   - /pg-cleaner/.gitlab-ci.yml
   - /puppeteer/.gitlab-ci.yml
   - /wait-for-postgres/.gitlab-ci.yml
+  - /fluentd-modsecurity/.gitlab-ci.yml
```
Comment (douglasduteil) [inline .gitlab-ci.yml]: Alphabetic order please :)

File: README.md [hunk 2]
```diff
@@ -31,19 +31,21 @@ $ docker pull docker.pkg.github.com/socialgouv/docker/<image>:<version>
...
+| **kubectl**              | `docker pull registry.gitlab.factory.social.gouv.fr/socialgouv/docker/kubectl:0.29.0`             | [![README](https://img.shields.io/badge/README--green.svg)](./kubectl/README.md)            |
+| **nginx4spa**            | `docker pull registry.gitlab.factory.social.gouv.fr/socialgouv/docker/nginx4spa:0.29.0`           | [![README](https://img.shields.io/badge/README--green.svg)](./nginx4spa/README.md)          |
+| **puppeteer**            | `docker pull registry.gitlab.factory.social.gouv.fr/socialgouv/docker/puppeteer:0.29.0`           | [![README](https://img.shields.io/badge/README--green.svg)](./puppeteer/README.md)          |
+| **wait_for_postgres**    | `docker pull registry.gitlab.factory.social.gouv.fr/socialgouv/docker/wait_for_postgres:0.29.0`   | [![README](https://img.shields.io/badge/README--green.svg)](./wait_for_postgres/README.md)  |
+| **fluentd-modsecurity**  | `docker pull registry.gitlab.factory.social.gouv.fr/socialgouv/docker/fluentd-modsecurity:0.29.0` | [![README](https://img.shields.io/badge/README--green.svg)](./fluent-modsecurity/README.md) |
```
Comment (douglasduteil) [inline README.md]: Alphabetic order please :)

File: fluentd-modsecurity/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+ && apt-get install -y --no-install-recommends make=4.2.1-1.2 gcc=4:8.3.0-1 g++=4:8.3.0-1 \
+ && gem install fluent-plugin-elasticsearch:4.0.7 \
+ && gem install fluent-plugin-modsecurity:0.1.4 \
+ && gem sources --clear-all \
+ && SUDO_FORCE_REMOVE=yes \
+    apt-get purge -y --auto-remove \
+                  -o APT::AutoRemove::RecommendsImportant=false \
+                  $buildDeps \
+ && rm -rf /var/lib/apt/lists/* \
+ && rm -rf /tmp/* /var/tmp/* /usr/lib/ruby/gems/*/cache/*.gem
```
Comment (douglasduteil) [inline fluentd-modsecurity/Dockerfile]: Prefer using 
- `set -ex`: so we can see the in between steps of the run
- `apk add --no-cache --virtual xxx`: so you can install without cache and `apk del xxx` the bin installed for the build only
https://github.com/SocialGouv/docker/blob/v0.29.0/azure-db/Dockerfile#L10-L29

File: fluentd-modsecurity/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+WORKDIR /plugins/fluentd-modsecurity
+RUN gem build fluent-plugin-modsecurity.gemspec && \
+    fluent-gem install ./fluent-plugin-modsecurity-0.2.gem
```
Comment (douglasduteil) [inline fluentd-modsecurity/Dockerfile]: Can you group your RUN ?

File: fluentd-modsecurity/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,37 @@
+FROM alpine/git:v2.24.1 as builder
+
+WORKDIR /
+RUN git clone https://github.com/bitsofinfo/fluentd-modsecurity.git
+
+FROM fluent/fluentd:v1.9.3-debian-1.0
+
+# Use root account to use apt
+USER root
```
Comment (douglasduteil) [inline fluentd-modsecurity/Dockerfile]: Is `fluent/fluentd:v1.9.3-debian-1.0` using another user ? If yes, can you document it ?

Comment (Virasack) [reply fluentd-modsecurity/Dockerfile]: Fait

File: fluentd-modsecurity/Dockerfile (hunk 4 above)
Comment (Virasack) [reply fluentd-modsecurity/Dockerfile]: fait

File: fluentd-modsecurity/Dockerfile (hunk 3 above)
Comment (Virasack) [reply fluentd-modsecurity/Dockerfile]: - set -ex -> OK
- apk add --no-cache --virtual xxx -> On utilise une Debian donc pas d' "apk". Les commandes "apt-get update", "apt-get purge" et "rm -rf /var/lib/apt/lists/*" sont équivalentes .

Comment (Virasack) [reply fluentd-modsecurity/Dockerfile]: Test CI -> OK

File: fluentd-modsecurity/.gitlab-ci.yml [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    CONTEXT: fluentd-modsecurity
+    IMAGE_NAME: ${CI_REGISTRY_IMAGE}/fluentd-modsecurity
+
+Test fluentd-modsecurity:
+  stage: "Test"
+  needs:
+    - Lint helm
+    - Build helm
+  image: "${CI_REGISTRY_IMAGE}/fluentd-modsecurity:${CI_COMMIT_SHA}"
+  environment: feature-dev
```
Comment (douglasduteil) [inline fluentd-modsecurity/.gitlab-ci.yml:22]: Does the `Test fluentd-modsecurity` require a connection to the cluster ?

Comment (Virasack) [reply fluentd-modsecurity/.gitlab-ci.yml:22]: Non on peut le supprimer

**Pull request merged**